test/e2e_node/oomkiller_linux_test.go

```diff
@@ -120,18 +136,33 @@ func runOomKillerTest(f *framework.Framework, testCase testCase, kubeReservedMem
 			e2epod.NewPodClient(f).Create(context.TODO(), testCase.podSpec)
 		})
 
-		ginkgo.It("The containers terminated by OOM killer should have the reason set to OOMKilled", func() {
-			ginkgo.By("Waiting for the pod to be failed")
-			err := e2epod.WaitForPodTerminatedInNamespace(context.TODO(), f.ClientSet, testCase.podSpec.Name, "", f.Namespace.Name)
-			framework.ExpectNoError(err, "Failed waiting for pod to terminate, %s/%s", f.Namespace.Name, testCase.podSpec.Name)
+		if testCase.expectPodRunning {
+			ginkgo.It("The containers should not be OOMKilled", func() {
+				gomega.Consistently(context.TODO(), func(ctx context.Context) error {
+					pod, err := f.ClientSet.CoreV1().Pods(testCase.podSpec.Namespace).Get(ctx, testCase.podSpec.Name, metav1.GetOptions{})
+					if err != nil {
+						return fmt.Errorf("expected the pod %s to exist: %w", pod.Name, err)
+					}
+					if pod.Status.Phase != v1.PodRunning {
+						return fmt.Errorf("pod %s: unexpected status %s, expected status: %s", pod.Name, pod.Status.Phase, v1.PodPending)
+					}
+					return nil
+				}, 10*time.Second, 200*time.Millisecond).Should(gomega.BeNil())
```

Comment:
```suggestion
				}, 10*time.Second, f.Timeouts.Poll).Should(gomega.BeNil())
```

Let's keep it consistent with other tests. 2 seconds is fast enough. 200 millisecond sounds excessive

Reply:
Make sense!